### PR TITLE
windows: add module to install the wintun.dll by embedding it in the client's exe and then copying it out at runtime

### PR DESCRIPTION
**Pull request by ReactorScram**

### CODE REVIEW

**Comment by ReactorScram** [comment]
I'll say this one's ready for review if the CI passes. There's more I could do with it, but if it compiles I might as well get it into main

File: rust/windows-client/src-tauri/src/wintun_install.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    // SHA256 e5da8447dc2c320edc0fc52fa01885c103de8c118481f683643cacc3220dafce
+    Some(_DllBytes {
+        bytes: include_bytes!("../../wintun/bin/amd64/wintun.dll"),
+        expected_sha256: "e5da8447dc2c320edc0fc52fa01885c103de8c118481f683643cacc3220dafce",
+    })
+}
+
+#[cfg(target_arch = "aarch64")]
+fn _get_platform_dll_bytes() -> Option<&'static [u8]> {
+    // wintun supports aarch64 but it's not in the Firezone repo yet
```
Comment (jamilbk) [inline rust/windows-client/src-tauri/src/wintun_install.rs:26]: We did some customer discovery on this -- apparently there's no demand (yet) for windows aarch64 from our customer base. I imagine there will be eventually, so this approach is good for now.

File: rust/windows-client/src-tauri/src/wintun_install.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+//! "Installs" wintun.dll at runtime by copying it into whatever folder the exe is in
+
+pub(crate) struct _DllBytes {
+    /// Bytes embedded in the client with `include_bytes`
+    bytes: &'static [u8],
+    /// Expected SHA256 hash
```
Comment (jamilbk) [inline rust/windows-client/src-tauri/src/wintun_install.rs:6]: 👍

**Comment by ReactorScram** [comment]
I had the wrong base branch. This is blocked on PR #2842

**Review by conectado — APPROVED** [review]
LGTM!

**Pull request merged**